integrations/access/accessrequest/app.go

```diff
@@ -389,6 +475,43 @@ func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest)
 	return recipientSet.ToSlice()
 }
 
+func (a *App) recipientsFromAccessMonitoringRules(ctx context.Context, req types.AccessRequest) *common.RecipientSet {
+	log := logger.Get(ctx)
+	recipientSet := common.NewRecipientSet()
+
+	// This switch is used to determine which plugins we are enabling access monitoring notification rules for.
+	switch a.pluginType {
+	// Enabled plugins are added to this case.
+	case types.PluginTypeSlack:
+		log.Debug("Applying access monitoring rules to request")
+	default:
+		return &recipientSet
+	}
+
+	a.accessMonitoringRules.RLock()
+	defer a.accessMonitoringRules.RUnlock()
+
+	for _, rule := range a.accessMonitoringRules.rules {
```

Comment:
Do we need to hold the lock over the entire "for" loop, or just to fetch the recipients? In the latter case, I would just make a helper method that returns them under lock:

```go
func (a *App) getAccessMonitoringRules() []xxx {
  a.accessMonitoringRules.RLock()
  defer a.accessMonitoringRules.RUnlock()
  return a.accessMonitoringRules.rules
}
```

And then just call it here:

```go
for _, rule := range a.getAccessMonitoringRules() {
...
}
```

Reply:
Since we are dealing with a map we must clone it under lock and return it. That's the only viable way


```
func (a *App) getAccessMonitoringRules() []xxx { 
a.accessMonitoringRules.RLock() 
defer a.accessMonitoringRules.RUnlock() 
return maps.Clone(a.accessMonitoringRules.rules)
 }
``